duckduckgo/BrowserServicesKit
# Fix syncing empty favorites folders

Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/414235014887631/1205843304285892/f
iOS PR: https://github.com/duckduckgo/iOS/pull/2121
macOS PR: https://github.com/duckduckgo/macos-browser/pull/1791
What kind of version bump will this require?: Patch

**Description**:
Update bookmarks sync response handler to actually process favorites
also when an empty folder is received from the server.



**Steps to test this PR**:
1. Connect 2 devices in Sync. Populate favorites.
2. Remove all favorites on one device (it's easiest on macOS).
3. Verify that all favorites are removed from another device after sync completes.



**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
